lib/list_versions.go

```diff
@@ -183,34 +196,31 @@ func removeDuplicateVersions(elements []string) []string {
 	return result
 }
 
-// validVersionFormat : returns valid version format
-/* For example: 0.1.2 = valid
-// For example: 0.1.2-beta1 = valid
-// For example: 0.1.2-alpha = valid
-// For example: a.1.2 = invalid
-// For example: 0.1. 2 = invalid
-*/
-func validVersionFormat(version string) bool {
-	// Getting versions from body; should return match /X.X.X-@/ where X is a number,@ is a word character between a-z or A-Z
-	// Follow https://semver.org/spec/v1.0.0-beta.html
-	// Check regular expression at https://rubular.com/r/ju3PxbaSBALpJB
-	semverRegex := regexp.MustCompile(`^(\d+\.\d+\.\d+)(-[a-zA-z]+\d*)?$`)
-	return semverRegex.MatchString(version)
-}
+// validVersionFormat : Return true if valid semantic version provided based on the type of version requested
+func validVersionFormat(version string, validation ...*regexp.Regexp) bool {
+	var semverRegex *regexp.Regexp
+
+	switch len(validation) {
+	case 0:
+		semverRegex = regexSemVer.Full
+	case 1:
+		semverRegex = validation[0]
+		// Limit the valid values in this switch case to only minor and patch versions,
+		// as currently it is used in `ShowLatestImplicitVersion()` only
+		validValues := []*regexp.Regexp{regexSemVer.Minor, regexSemVer.Patch}
+		if !slices.Contains(validValues, semverRegex) {
+			logger.Fatalf("Internal error: invalid \"validation\" argument value")
+		}
+	default:
+		logger.Fatalf("Internal error: invalid number of arguments (must be 1 or 2, got %d)", 1+len(validation))
+	}
 
-// ValidMinorVersionFormat : returns valid MINOR version format
-/* For example: 0.1 = valid
-// For example: a.1.2 = invalid
-// For example: 0.1.2 = invalid
-*/
-func validMinorVersionFormat(version string) bool {
-	// Getting versions from body; should return match /X.X./ where X is a number
-	semverRegex := regexp.MustCompile(`^(\d+\.\d+)$`)
+	semverRegex = regexp.MustCompile(`^` + semverRegex.String() + `$`)
```

Comment:
It's a shame that the original versions can't include the match of start/end, and require Compile(Compile().String()), but oh well, I think it's definitely safer as is :)